File: BackEnd/src/main/java/com/Library/lmsproject/service/ReviewService.java
package com.library.lmsproject.service;

import com.library.lmsproject.dto.request.CreateReviewRequestDTO;
import com.library.lmsproject.dto.request.UpdateReviewRequestDTO;
import com.library.lmsproject.dto.response.ReviewResponseDTO;

import java.util.List;

public interface ReviewService {

    ReviewResponseDTO createReview(CreateReviewRequestDTO request, Long userId);

    List<ReviewResponseDTO> getReviewsByBook(Long bookId);

    void deleteReview(Long reviewId, Long userId);

    List<ReviewResponseDTO> getAllReviews();

    ReviewResponseDTO updateReview(Long reviewId, UpdateReviewRequestDTO request, Long userId);

}

File: FrontEnd/src/services/reviewService.js
import axiosClient from "../lib/axiosClient";

// GET - Admin lấy tất cả reviews
export const getAllReviews = async () => {
  const response = await axiosClient.get("/api/reviews/getAllReviews");
  return response.data;
};

// GET - Lấy danh sách reviews theo bookId
export const getReviewsByBook = async (bookId) => {
  const response = await axiosClient.get(`/api/reviews/book/${bookId}`);
  return response.data;
};

// POST - Tạo review mới
export const createReview = async (userId, reviewData) => {
  const response = await axiosClient.post("/api/reviews/createReview", reviewData, {
    params: { userId }
  });
  return response.data;
};

// PUT - Cập nhật review
export const updateReview = async (reviewId, userId, reviewData) => {
  const response = await axiosClient.put(`/api/reviews/updateReview/${reviewId}`, reviewData, {
    params: { userId }
  });
  return response.data;
};


// DELETE - Xóa review (chỉ owner mới được xóa)
export const deleteReview = async (reviewId, userId) => {
  const response = await axiosClient.delete(`/api/reviews/deleteReview/${reviewId}`, {
    params: { userId }
  });
  return response.data;
};
